pandas/tests/util/test_assert_numpy_array_equal.py

```diff
@@ -175,3 +177,24 @@ def test_numpy_array_equal_copy_flag(other_type, check_same):
             tm.assert_numpy_array_equal(a, other, check_same=check_same)
     else:
         tm.assert_numpy_array_equal(a, other, check_same=check_same)
+
+
+def test_numpy_array_equal_contains_na():
+    # https://github.com/pandas-dev/pandas/issues/31881
+    a = np.array([True, False])
+    b = np.array([True, NA])
+
+    msg = """numpy array are different
+
+numpy array values are different \\(50.0 %\\)
+\\[left\\]:  \\[True, False\\]
+\\[right\\]: \\[True, <NA>\\]"""
+
+    with pytest.raises(AssertionError, match=msg):
+        tm.assert_numpy_array_equal(a, b)
+
+
+def test_numpy_array_equal_identical_na(nulls_fixture):
+    a = np.array([nulls_fixture])
+
+    tm.assert_numpy_array_equal(a, a)
```

Comment:
Different NA-likes are not considered equal? Eg `np.array([np.nan])` vs `np.array([pd.NA])` ? 

I think for now it would be good to keep those not equal (since they behave differently), but if so, we should best add a test for that as well.

Reply:
Yes, those are comparing unequal, will add a test